### PR TITLE
Fix bunx command for github package #5974

### PR DESCRIPTION
**Pull request by axlEscalada**
### What does this PR do?
Fix the `bunx` command when a github repo is passed as an argument. There is a bug where the package name is parsed as an `UpdateRequest` that leaves the name of the package empty and fails when trying to concatenate `name + @ + version`
In bunx_command name and version are checked and if they are equal set the name as the package name, and format it without `@`. Also this fmt package is used to build the path where is installed.

Fixes: https://github.com/oven-sh/bun/issues/5974

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

- [ ] I checked the lifetime of memory allocated to verify it's (1) freed and (2) only freed when it should be
- [x] I or my editor ran `zig fmt` on the changed files
- [x] I included a test for the new code, or an existing test covers it
- [ ] JSValue used outside outside of the stack is either wrapped in a JSC.Strong or is JSValueProtect'ed

### CODE REVIEW

File: src/cli/bunx_command.zig [hunk 1]
```diff
@@ -262,22 +262,34 @@ pub const BunxCommand = struct {
         else
             update_request.version.literal.slice(update_request.version_buf);
 
+        var package_fmt: []const u8 = undefined;
```
Comment (Jarred-Sumner) [inline src/cli/bunx_command.zig]: `undefined` is dangerous and should be avoided whenever possible

```suggestion
        const package_fmt: []const u8 = brk: {
  if (!strings.eql(update_request.version_buf, update_request.name)) {
            break :brk try std.fmt.allocPrint(
                ctx.allocator,
                "{s}@{s}",
                .{
                    update_request.name,
                    display_version,
                },
            );
        }
        
        break :brk update_request.name;
};
```

**Comment by Jarred-Sumner** [comment]
Thank you for this. One nitpicky comment

File: src/cli/bunx_command.zig (hunk 1 above)
Comment (axlEscalada) [reply src/cli/bunx_command.zig]: fixed

**Comment by Electroid** [comment]
Tested and good-to-go. Thanks!

**Pull request merged**